mlos_bench/mlos_bench/util.py

```diff
@@ -100,3 +71,27 @@ def check_required_params(config: Dict[str, Any], required_params: Iterable[str]
         raise ValueError(
             "The following parameters must be provided in the configuration"
             + f" or as command line arguments: {missing_params}")
+
+
+def get_git_info(path: str = ".") -> Tuple[str, str]:
+    """
+    Get the git repository and commit hash of the current working directory.
+
+    Parameters
+    ----------
+    path : str
+        Path to the git repository.
+
+    Returns
+    -------
+    (git_repo, git_commit) : Tuple[str, str]
+        Git repository URL and last commit hash.
+    """
+    git_repo = subprocess.check_output(
+        ["cd", path, "&&", "git", "remote", "get-url", "origin"],
+        shell=True, text=True).strip()
+    git_commit = subprocess.check_output(
+        ["cd", path, "&&", "git", "rev-parse", "HEAD"],
```

Comment:
This may not work for Windows (though not sure if we care anymore).

Tests for it?

Reply:
I agree, we need unit tests for it. (It seems to work for both Windows and Linux, btw)

Reply:
The `&&` syntax was the main part I wasn't sure about for `pwsh` vs `powershell` vs `cmd`.
However, you could do something like `["git", "-C", path, "rev-parse", "HEAD"]` to use git's builtin ability to move to a target directory, which should be cross platform and avoid extra fork/exec issues.

Reply:
great point! I'll fix the code and add unit tests